### PR TITLE
Add feautre to perform daily reboot all nodes in dev environment

### PR DESCRIPTION
**Pull request by zeroalphat**
Add systemd service and timer for triggering daily reboot all nodes on dev environment.

### CODE REVIEW

File: debian/usr/bin/trigger-reboot-all-nodes [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+NECO_COMMAND="/usr/bin/neco"
+CKECLI_COMMAND="/usr/bin/ckecli"
+YES_COMMAND="/usr/bin/yes"
+
+ENVIROMENT=$(${NECO_COMMAND} config get env)
```
Comment (masa213f) [inline debian/usr/bin/trigger-reboot-all-nodes:6]: Could you do early return?

```bash
ENVIROMENT=$(${NECO_COMMAND} config get env)
if [ "$ENVIROMENT" != "dev" ]; then
    echo "This script is only for dev environment"
    exit 0
fi

REBOOT_LIST=$(${NECO_COMMAND} rebooter list)
REBOOT_QUEUE=$(${CKECLI_COMMAND} rq list)
# Confirm that the reboot list and reboot queue are empty
if [ "$REBOOT_LIST" != "null" ] || [ "$REBOOT_QUEUE" != "null" ]; then 
    echo "The reboot list or reboot queueu is not empty. Skipping reboot all nodes"
    exit 0
fi

$NECO_COMMAND rebooter disable
sh -c $YES_COMMAND | $NECO_COMMAND rebooter reboot-worker --without-state retiring,retired
$NECO_COMMAND rebooter enable
```

File: debian/usr/bin/trigger-reboot-all-nodes [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+if [ "$ENVIROMENT" = "dev" ]; then
+    REBOOT_LIST=$(${NECO_COMMAND} rebooter list)
+    REBOOT_QUEUE=$(${CKECLI_COMMAND} rq list)
+   
+    # Confirm that the reboot list and reboot queue are empty
+    if [ "$REBOOT_LIST" = "null" ] && [ "$REBOOT_QUEUE" = "null" ]; then 
+        $NECO_COMMAND rebooter disable
+        sh -c $YES_COMMAND | $NECO_COMMAND rebooter reboot-worker --without-state retiring,retired
+        $NECO_COMMAND rebooter enable
```
Comment (masa213f) [inline debian/usr/bin/trigger-reboot-all-nodes:15]: Could you run these command on only one boot server?
I think we can use `neco rebooter leader` or `ckecli leader`.

**Review by masa213f — CHANGES_REQUESTED** [review]
LGTM

File: debian/usr/bin/trigger-reboot-all-nodes (hunk 2 above)
Comment (zeroalphat) [reply debian/usr/bin/trigger-reboot-all-nodes]: I failed to take into consideration that it works on all boot servers.
I added a process to check if the server the script is running on is a leader.

File: debian/usr/bin/trigger-reboot-all-nodes [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+LEADER=$(${NECO_COMMAND} rebooter leader)
+if [ "$HOSTNAME" != "$LEADER" ]; then
+    echo "This host is not a leader. Skipping reboot all nodes"
+    exit 0
+fi
+
+REBOOT_LIST=$(${NECO_COMMAND} rebooter list)
+REBOOT_QUEUE=$(${CKECLI_COMMAND} rq list)
+# Confirm that the reboot list and reboot queue are empty
+if [ "$REBOOT_LIST" != "null" ] && [ "$REBOOT_QUEUE" != "null" ]; then
```
Comment (masa213f) [inline debian/usr/bin/trigger-reboot-all-nodes:24]: Is it correct? Should we change to `||`?

**Review by masa213f — CHANGES_REQUESTED** [review]

File: debian/usr/bin/trigger-reboot-all-nodes (hunk 3 above)
Comment (zeroalphat) [reply debian/usr/bin/trigger-reboot-all-nodes:24]: Thank you for pointing this out.
I had overlooked the need to change the conditional judgement 🙇

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**